src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -186,91 +192,112 @@ trait SlickExpressionParser extends JavaTokenParsers {
       // the optional entity rec is used for references. Since we know we are not dealing with a reference here
       // as this is the attribute final func, we can pass in None.
       val attributesOption = attributeQuery.unmarshalAttributes(wsIdAndAttributes.map((_, None))).get(context.workspaceContext.workspaceId)
-      attributesOption.map { attributes => Seq(attributes.getOrElse(attributeName, AttributeNull)) }.getOrElse(Seq.empty)
+      val wsExprResult = attributesOption.map { attributes => Seq(attributes.getOrElse(attributeName, AttributeNull)) }.getOrElse(Seq.empty)
+
+      //Return the value of the expression once for each entity we wanted to evaluate this expression against!
+      context.rootEntities.map( rec => (rec.name, wsExprResult) ).toMap
```

Comment:
The parser doesn't populate or join against the `EXPREVAL_TEMP` for workspace expressions, so we just find all the results and dupe them once per root entity in the batch here.